syncopy/nwanalysis/connectivity_analysis.py

```diff
@@ -378,33 +380,35 @@ def connectivityanalysis(data, method="coh", keeptrials=False, output="abs",
     # ----------------------------------------------------------------------------------
     # Sanitize output and call the chosen ComputationalRoutine on the averaged ST output
     # ----------------------------------------------------------------------------------
-
-    out = CrossSpectralData(dimord=st_dimord)
-
-    # now take the trial average from the single trial CR as input
-    av_compRoutine.initialize(st_out, out._stackingDim, chan_per_worker=None)
-    av_compRoutine.pre_check()   # make sure we got a trial_average
-    av_compRoutine.compute(st_out, out, parallel=kwargs.get("parallel"),
-                           log_dict=log_dict)
-
-    # attach potential older cfg's from the input
-    # to support chained frontend calls..
-    out.cfg.update(data.cfg)
-    # attach frontend parameters for replay
-    out.cfg.update({'connectivityanalysis': new_cfg})
-    return out
+    if method == 'csd':
+        new_cfg.update({'output': st_out.data.dtype.name})
+        st_out.cfg.update(data.cfg)
+        st_out.cfg.update({'cross_spectral': new_cfg})
```

Comment:
It's very important that the top-level keys inside a `cfg` are frontend meta function names, like `connectivityanalysis` or `freqanalysis`. This is how the `unwrap_cfg` decorator knows which part of a `cfg` needs to be applied to the current frontend call.

Reply:
done